module_fcst_grid_comp.F90

```diff
@@ -201,7 +203,37 @@ subroutine SetServicesNest(nest, rc)
       grid_typekind = ESMF_TYPEKIND_R8
     endif
 
-    if (trim(name)=="global") then
+    if (Atmos%grid_type==4) then
+       gridfile = "grid_spec.nc" ! default
+
+       if (open_file(fileobj, "INPUT/grid_spec.nc", "read")) then
+         if (variable_exists(fileobj, "atm_mosaic_file")) then
+           call read_data(fileobj, "atm_mosaic_file", gridfile)
+         endif
+         call close_file(fileobj)
+       endif
+
+      call ESMF_InfoGet(info, key="nx", value=nx, rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+      call ESMF_InfoGet(info, key="ny", value=ny, rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+      grid = ESMF_GridCreateNoPeriDim( regDecomp=(/layout(1),layout(2)/), &
+                                      minIndex=(/1,1/), &
+                                      maxIndex=(/nx,ny/), &
+                                      gridAlign=(/-1,-1/), &
+                                      coordSys=ESMF_COORDSYS_SPH_RAD, &
+                                      coordTypeKind=grid_typekind, &
+                                      decompflag=(/ESMF_DECOMP_SYMMEDGEMAX,ESMF_DECOMP_SYMMEDGEMAX/), &
+                                      name="fcst_grid", &
+                                      indexflag=ESMF_INDEX_DELOCAL, &  
+                                      rc=rc) 
```

Comment:
This block looks to be identical to the code in `else` block below. If it is indeed identical, these two cases should be combined into one.

Reply:
Logic has been combined for the two cases.